src/ios/CDVLocation.m

```diff
@@ -80,87 +81,111 @@ - (BOOL)isAuthorized
     return YES;
 }
 
-- (BOOL)isLocationServicesEnabled
-{
-    BOOL locationServicesEnabledInstancePropertyAvailable = [self.locationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 3.x
-    BOOL locationServicesEnabledClassPropertyAvailable = [CLLocationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 4.x
-
-    if (locationServicesEnabledClassPropertyAvailable) { // iOS 4.x
-        return [CLLocationManager locationServicesEnabled];
-    } else {
-        return NO;
-    }
+- (void)isLocationServicesEnabledWithCompletion:(void (^)(BOOL enabled))completion {
+    dispatch_async(queue, ^{
+        BOOL locationServicesEnabledClassPropertyAvailable = [CLLocationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 4.x
```

Comment:
Do we need this? The minimum version we support is 14, and this is available since 4.

Reply:
I didn't update the code itself, I only added the queue. However, I don't see any issues.

Reply:
You did remove the `locationServicesEnabledInstancePropertyAvailable` variable, which makes perfect sense in terms of code cleanse. The same could be applied to `locationServicesEnabledClassPropertyAvailable`.

Reply:
`locationServicesEnabledClassPropertyAvailable ` is being used a few lines below.

Reply:
I know. `respondsToSelector` checks whether an object can respond to a specific method selector at runtime. In this case, the check is used as a backward compatibility check across different iOS versions. But since this has been available since iOS 4, I believe it's unnecessary.

Reply:
Fixed.